opentelemetry-user-events-trace/src/trace/exporter.rs

```diff
@@ -0,0 +1,175 @@
+            Level::Informational,
+            Level::Verbose,
+        ];
+
+        let mut event_sets = vec![Arc::new(EventSet::new_unregistered())];
+
+        for &level in levels.iter() {
+            let event_set = eventheader_provider.register_set(level, keyword);
+            event_sets.push(event_set);
+        }
+        event_sets
+    }
+
+    pub(crate) fn export_span(&self, span: &SpanData) -> OTelSdkResult {
+        let level = Level::Informational; //TODO which level/tracepoint to use?
+        let event_set = self
+            .event_sets
+            .get(level.as_int() as usize)
+            .ok_or_else(|| OTelSdkError::InternalFailure("Failed to get event set".to_string()))?;
+
+        if event_set.enabled() {
+            let mut eb = EventBuilder::new();
+            eb.reset("Span", 0);
+            eb.opcode(Opcode::Info);
+            eb.add_value("__csver__", 1024, FieldFormat::UnsignedInt, 0);
+
+            eb.add_struct("PartA", 2, 0); // time, ext_dt
+            let datetime: DateTime<Utc> = span.end_time.into();
+            eb.add_str("time", datetime.to_rfc3339(), FieldFormat::Default, 0);
+            eb.add_struct("ext_dt", 2, 0);
+            eb.add_str(
+                "traceId",
+                span.span_context.trace_id().to_string(),
+                FieldFormat::Default,
+                0,
+            );
+            eb.add_str(
+                "spanId",
+                span.span_context.span_id().to_string(),
+                FieldFormat::Default,
+                0,
+            );
+
+            eb.add_struct("PartB", 5, 0);
+            eb.add_str("_typeName", "Span", FieldFormat::Default, 0);
+            eb.add_str("name", span.name.as_ref(), FieldFormat::Default, 0);
+            eb.add_str(
+                "parentId",
+                span.parent_span_id.to_string(),
+                FieldFormat::Default,
+                0,
+            );
+            let datetime: DateTime<Utc> = span.start_time.into();
+            eb.add_str("startTime", datetime.to_rfc3339(), FieldFormat::Default, 0);
+            eb.add_value(
+                "success",
+                matches!(span.status, Status::Ok | Status::Unset), // Check for Ok or Unset
+                FieldFormat::Boolean,
+                0,
+            );
+            eb.add_str(
+                "kind",
+                match span.span_kind {
+                    SpanKind::Internal => "internal",
+                    SpanKind::Server => "server",
+                    SpanKind::Client => "client",
+                    SpanKind::Producer => "producer",
+                    SpanKind::Consumer => "consumer",
+                },
+                FieldFormat::Default,
+                0,
+            );
+            eb.add_struct("PartC", span.attributes.len() as u8, 0);
+            for kv in span.attributes.iter() {
+                eb.add_str(
+                    kv.key.as_str(),
+                    kv.value.to_string(),
```

Comment:
We should be able to avoid `to_string()` for string-typed values. Refer to #187.

Reply:
Good point. Please see if the fix is as expected. Thanks.